common/CommonFunctions.ml

```diff
@@ -166,14 +156,56 @@ and get_published_info_class cls =
   get_first_release_string (get_first_release cls.obj_release.internal)
 
 and get_published_info_field field cls =
-  let expRelease = get_prototyped_release_string field.lifecycle in 
+  let expRelease = get_prototyped_release_string field.lifecycle in
   let clsRelease = get_first_release cls.obj_release.internal in 
   let fieldRelease =  get_first_release field.release.internal in
-  let clsReleaseIndex = list_index_of clsRelease DT.release_order in
-  let fieldReleaseIndex = list_index_of fieldRelease DT.release_order in
+  let clsReleaseIndex = list_index_of clsRelease code_name_order in
+  let fieldReleaseIndex = list_index_of fieldRelease code_name_order in
     if (not (expRelease = "")) then
       expRelease
     else if (not (clsReleaseIndex = -1)) && (not (fieldReleaseIndex = -1)) && (clsReleaseIndex < fieldReleaseIndex) then
       get_first_release_string fieldRelease
     else
       ""
+
+and render_template template_file json output_file =
+  let lines_read = Stdext.Unixext.read_lines template_file in
+  let templ = Mustache.of_string (String.concat "\n" lines_read) in
+  let rendered = Mustache.render templ json in
+  let out_chan = open_out output_file in
+  finally (fun () -> fprintf out_chan "%s" rendered)
+          (fun () -> close_out out_chan)
+
+let render_file (infile,outfile) json templates_dir dest_dir=
+  let input_path = Filename.concat templates_dir infile in
+  let output_path = Filename.concat dest_dir outfile in
+  render_template input_path json output_path
+
+let json_releases =
+  let json_of_rel x y = `O [
+    "code_name", `String (code_name_of_release_ext x);
+    "version_major", `Float (float_of_int x.version_major);
+    "version_minor", `Float (float_of_int x.version_minor);
+    "branding", `String x.branding;
+    "xs_branding", `String x.xs_branding;
+    "version_index", `Float (float_of_int y);
+    ]
+  in
+  let rec get_unique l =
+    match l with
+    | [] -> []
+    | hd::tl -> hd::(get_unique (List.filter (fun x -> not (x.version_major = hd.version_major && x.version_minor = hd.version_minor)) tl))
```

Comment:
I think splitting it in two makes it more readable
```ocaml
let deduplicate = List.filter (fun x -> not (x.version_major = hd.version_major && x.version_minor = hd.version_minor)
in hd :: get_unique (deduplicate tl)
```
or probably you can also use `Stedext.Listext.List.setify` (impl: https://github.com/xapi-project/stdext/blob/master/lib/listext.ml#L33)